src/plugins/intel_gna/src/transformations/utils/gather_sinking_utils.cpp

```diff
@@ -49,77 +51,167 @@ GatherInputsInfo GetFirstGatherInput(NodePtr node) {
     return GatherInputsInfo();
 }
 
-bool IfNodeHasGatherInputs(const Output<Node>& output) {
-    GatherInputsInfo inputs_info = GetFirstGatherInput(output.get_node_shared_ptr());
-    return !inputs_info.isEmpty();
+void swap_output_names(Output<Node> output1, Output<Node> output2) {
+    const auto node2_output_names = output2.get_names();
+    output2.set_names(output1.get_names());
+    output1.set_names(node2_output_names);
 }
 
-namespace {
+void swap_friendly_names(NodePtr node1, NodePtr node2) {
+    const std::string node2_name = node2->get_friendly_name();
+    node2->set_friendly_name(node1->get_friendly_name());
+    node1->set_friendly_name(node2_name);
+}
 
-bool HasDynamicRankInput(NodePtr node) {
-    for (const auto& input_node : node->input_values()) {
-        const Rank output_rank = input_node.get_partial_shape().rank();
-        if (output_rank.is_dynamic())
-            return true;
-    }
-    return false;
+void swap_names(NodePtr node1, NodePtr node2) {
+    swap_friendly_names(node1, node2);
+    swap_output_names(node1->output(0), node2->output(0));
 }
 
-Rank::value_type GetMaxInputRank(const NodePtr& node) {
-    Rank::value_type max_input_rank = 0;
-    for (auto& input_node : node->input_values()) {
-        const Rank output_rank = input_node.get_partial_shape().rank();
-        if (output_rank.is_dynamic())
-            return -1;
-        const Rank::value_type output_rank_len = output_rank.get_length();
-        if (output_rank_len > max_input_rank)
-            max_input_rank = output_rank_len;
+namespace sink_forward {
+/** @brief
```

Comment:
Are you sure you want to have such long @brief
Following the documentation
https://www.doxygen.nl/manual/commands.html#cmdbrief

" A brief description may span several lines (although it is advised to keep it brief!). A brief description ends when a blank line or another sectioning command is encountered."